primitives/tracing/src/types.rs

```diff
@@ -608,11 +507,17 @@ mod std_features {
 			.iter()
 			.map(|f| (f.name(), f))
 			.collect();
+		let mut vl = attrs.fields.0;
 
-		let values = attrs.fields.0.iter().filter_map(|(f, v)| {
+		let values = vl.iter_mut().filter_map(|(f, v)| {
 			if let Ok(inner) = std::str::from_utf8(&f.0) {
 				if let Some(field) = fields.get(inner) {
-					return Some((field, v.as_ref().map(|d| d.as_value()).flatten()))
+					return Some((field,
```

Comment:
could reduced to `return Some((
   field, 
   v.as_mut().and_then(|d| d.as_value())
))
`